cluster/src/main/scala/io/snappydata/cluster/ExecutorInitiator.scala

```diff
@@ -191,10 +196,12 @@ object ExecutorInitiator extends Logging {
                     rpcenv.setupEndpoint("Executor", executor)
                   }
                   prevDriverURL = url
+                  testLock.synchronized(testLock.notify())
                 case None =>
                   // If driver url is none, already running executor is stopped.
                   prevDriverURL = ""
               }
+              setRetryFlag(false)
```

Comment:
Can you tell what moving this setting does?

Reply:
When multiple threads get OOM exceptions every task tries to set the flag to true.  Hence sometimes the executor is stopped & started multiple times because of the loop condition. This was not harmful till now, and in actual cases very rare.  Moving set flag  does not avoid this. However it reduces the occurrences.